packages/field/src/FormatMixin.js

```diff
@@ -171,9 +171,26 @@ export const FormatMixin = dedupeMixin(
       }
 
       __callParser(value = this.formattedValue) {
-        let result;
-        if (typeof value === 'string') {
-          result = this.parser(value, this.formatOptions);
+        if (value === '') {
+          // Ideally, modelValue should be undefined for empty strings.
+          // For backwards compatibility we return an empty string.
+          return '';
+        }
+
+        if (typeof value !== 'string') {
+          // This means there is nothing to find inside the view that can be of
+          // interest to the Application Developer or needed to store for future form state
+          // retrieval.
+          return undefined;
+        }
+
+        const result = this.parser(value, this.formatOptions);
+        if (!result) {
```

Comment:
I feel like we're doing double work here. We return a new Unparseable if the result is falsy, and then we have another check that returns a new Unparseable of result is if type 'undefined', which would already be caught by the falsy check above. We should combine the conditions into 1 conditional.

Reply:
That's what happens when you have to cherry-pick commits you worked on two weeks ago.
In this branch everything was still right: https://github.com/ing-bank/lion/blob/fix/formIssues/packages/field/src/FormatMixin.js#L219

Good catch though